### ISSUE TITLE
error: static assertion failed: Meta Types must be fully defined

### ISSUE THREAD

**Issue by jakiki6** [blueprint-freespeech/ricochet-refresh#196]
I'm currently porting this to GNU Guix but compiling it fails with the following error log:

[guix.log](https://github.com/user-attachments/files/17264144/guix.log)

**Comment by morganava** [comment]
Hard to tell without diving into your build environment and Qt6 specifics. Official releases are build with Qt5, so perhaps the source itself needs to be updated to work with Qt6?

**Comment by jakiki6** [comment]
okay so building with Qt5 seems to work now, thanks

**Comment by morganava** [comment]
Official release build scripts can be found here in the ricochet-build repo:
- https://github.com/blueprint-freespeech/ricochet-build/blob/main/projects/ricochet-refresh/build

**Comment by jakiki6** [comment]
https://github.com/jakiki6/lauras-channel/blob/add7f108644c5784e017322e88641233629bcaf1/laura/packages/utils.scm#L847

**Comment by morganava** [comment]
New version will use wxWidgets most likely

**Comment by odiferousmint** [comment]
> New version will use wxWidgets most likely

New version of Ricochet? How come? Have you guys considered all the other lightweight candidates? I think I am fine with wxWidgets though.

**Comment by morganava** [comment]
@odiferousmint yeah v4 will (most likely) rely on wxWidgets for (more) native platform widgets as it seems to be the least bad option. Are there other lightweight UI toolkit alternatives available? The constraints we're trying to satisfy:
- native widgets (e.g. win32 on windows, cocoa/whatever it is now on macOS, gtk or qt on linux, etc)
- translation and localisation support (especially RTL locales)
- accessibility tech support (e.g. screen readers, high contrast based on system settings, etc)
- no electron/webview/whatever stacks (a security nightmare re user deanonimisation via proxy bypass)
